# Commands for Week 5

See full Week 5 instructions on [Protocals.io](https://www.protocols.io/private/bba6f8f98e03c9de92641ca3810b212c)

---
## Getting Started

> This week in lab we’re going to learn how to make phylogenetic trees. We’ll start by creating a multiple sequence alignment with MUSCLE, and then we will make bootstrapped maximum likelihood phylogenetic trees with RAxML. We’ll use the Newick files generated by RAxML to visualize trees in an online tree visualization tool called the Interactive Tree of Life (iToL). We’ll do this using toy datasets, and then try out some trees on your project datsets.

### Get into liverpool

```bash
ssh michelsd@liverpool.its.carleton.edu`
```

### Copy data to the class shared site

**!! STILL NEED TO DO !!**

```bash
cp ERR599031_ORFs.faa /usr/local/data/class_shared/
cp  ERR598983_interproscan_results.txt /usr/local/data/class_shared/
```



---

## A. Aligning Sequences

### Get new toy data

Copy the toy dataset from the data directory to new toy dataset directory

```bash
mkdir alignments_and_trees
cd alignments_and_trees/
cp /usr/local/data/toy_datasets/toy_dataset_PSII_protein.faa .
```

## Make a multiple sequence alignment using muscle

```bash
muscle -in toy_dataset_PSII_protein.faa -out toy_dataset_PSII_protein_aligned.afa
```

> *What this means:* muscle is the name of the program
>
> * `–in` defines the name of your input file, which can be either DNA or protein
> * `-out` defines the name of your output file. I like to give them an easy-to-recognize name with the extension “.afa”, which stands for “aligned fasta file.”

### Look at alignment with Seaview

To secure copy file the file:

- Open a new terminal window and don't log into liverpool
- Navigate to the right folder

```bash
scp michelsd@liverpool.its.carleton.edu:/Accounts/michelsd/toy_dataset_directory/alignments_and_trees/toy_dataset_PSII_protein_aligned.afa .
```

Open with application called "Seaview" by dragging file onto window.

> Seaview shows the names of the sequences to the left. The letters to the right are the amino acids in your sequence, color-coded to make the alignment easier to see. You can easily see that some regions of the sequence are more highly conserved than others, and that some species appear to have an insertion or deletion in specific regions of the sequence. Note that this is an **amino acid** alignment, not a nucleotide alignment. (You could easily use muscle to align in nucleotides as well.)



---

## B. Creating phylogenetic trees with RAxML

Use Rika's python script to convert from  aligned fasta (.afa) file into Phylip file (.phy)

```bash
convert_afa_to_phy.py toy_dataset_PSII_protein_aligned.afa
```

> The top of a Phylip file indicates the number of sequences (in this case, 17) and the number of base pairs in the aligned sequences (571). It’s followed by the first 54 bases of each sequence, broken up by a space every 10 letters. Every new line shows a new sequence (from a different species or organism). The next 54 letters of each sequence starts after the first set, and continues like this for the rest of the file. (Yes, it’s weird, but a lot of tree-building programs use this format.)



Make a tree with raxml

```
raxmlHPC-PTHREADS-AVX -f a -# 20 -m PROTGAMMAAUTO -p 12345 -x 12345 -s toy_dataset_PSII_protein_aligned.phy -n toy_dataset_PSII_protein.tree -T 4
```

> What this means:
>
> * `raxmlHPC-PTHREADS-AVX` is the name of the software package. This one is configured for the processors that are specific to this server.
> * `-f a` allows for rapid bootstrapping. This means RAxML will do a maximum likelihood search based on your protein sequences, and then bootstrap it as many times as you wish.
> * `-# 20` tells the program to bootstrap 20 times.
> * `-m PROTGAMMAAUTO` tells the program that these are protein sequences, and tells the program how to model protein evolution. To get into this is beyond the scope of this class, but fortunately RAxML is able to automatically choose the best one for us based on the number of sequences and the type of data we have.
> * `-p` and `–x` provide seed numbers so that the program can generate random numbers for the bootstrapping process.
> * `-s` gives the name of your input Phylip file
> * `-n` gives the name of your output Newick file, which will be made into a tree.
> * `-T` determines the number of threads. This is sort of like determining how many processors you'll use up for this process. Today, we'll use 4. Please don't use more than this without asking first.