# Commands for Week 5

See full Week 5 instructions on [Protocals.io](https://www.protocols.io/private/bba6f8f98e03c9de92641ca3810b212c)

---
## START

> This week in lab we’re going to learn how to make phylogenetic trees. We’ll start by creating a multiple sequence alignment with MUSCLE, and then we will make bootstrapped maximum likelihood phylogenetic trees with RAxML. We’ll use the Newick files generated by RAxML to visualize trees in an online tree visualization tool called the Interactive Tree of Life (iToL). We’ll do this using toy datasets, and then try out some trees on your project datsets.

### Get into liverpool

```bash
ssh michelsd@liverpool.its.carleton.edu`
```

### Copy data to the class shared site

**!! STILL NEED TO DO !!**

```bash
cp ERR599031_ORFs.faa /usr/local/data/class_shared/
cp  ERR598983_interproscan_results.txt /usr/local/data/class_shared/
```

### Get new toy data

Copy the toy dataset from the data directory to new toy dataset directory

```bash
mkdir alignments_and_trees
cd alignments_and_trees/
cp /usr/local/data/toy_datasets/toy_dataset_PSII_protein.faa .
```

---
## A. ALIGNING SEQUENCES

### Make a multiple sequence alignment using muscle

```bash
muscle -in toy_dataset_PSII_protein.faa -out toy_dataset_PSII_protein_aligned.afa
```

> *What this means:* muscle is the name of the program
> 
> * `–in` defines the name of your input file, which can be either DNA or protein
> * `-out` defines the name of your output file. I like to give them an easy-to-recognize name with the extension “.afa”, which stands for “aligned fasta file.”

### Look at alignment with Seaview

To secure copy file the file:

- Open a new terminal window and don't log into liverpool
- Navigate to the right folder

```shell
scp michelsd@liverpool.its.carleton.edu:/Accounts/michelsd/toy_dataset_directory/alignments_and_trees/toy_dataset_PSII_protein_aligned.afa .
```


